pkg/volume/cinder/attacher.go

```diff
@@ -124,7 +124,18 @@ func (attacher *cinderDiskAttacher) VolumesAreAttached(specs []*volume.Spec, nod
 		volumesAttachedCheck[spec] = true
 		volumeSpecMap[volumeSource.VolumeID] = spec
 	}
-	attachedResult, err := attacher.cinderProvider.DisksAreAttached(volumeIDList, string(nodeName))
+	instances, res := attacher.cinderProvider.Instances()
+	if !res {
+		return volumesAttachedCheck, fmt.Errorf("underlying cloud provider does not claim to support Instances api")
+	}
+	instanceid, err := instances.InstanceID(nodeName)
+	if err != nil {
+		return volumesAttachedCheck, err
+	}
+	if ind := strings.LastIndex(instanceid, "/"); ind >= 0 {
```

Comment:
nodeName is the nova name (used as hostname), instanceId is its uuid. For the / I think it is indeed useless. Like you said I basically copy/paste the working code from Attach ;)